packages/core/src/util/renderer.ts

```diff
@@ -656,9 +656,7 @@ export const mapStateToLayoutProps = (
   const visible: boolean = has(ownProps, 'visible')
     ? ownProps.visible
     : isVisible(ownProps.uischema, rootData, ownProps.path);
-  const enabled: boolean = has(ownProps, 'enabled')
-    ? ownProps.enabled
-    : isEnabled(ownProps.uischema, rootData, ownProps.path);
+  const enabled: boolean = isEnabled(ownProps.uischema, rootData, ownProps.path);
```

Comment:
`isEnabled` only checks locally for enablement, e.g. whether the given ui schema has a rule which enables / disables it. This will lead to unexpected behavior in case of nested layouts where only the outer most layout is disabled via a rule. In general the most specific rule should win.